deployment/ccip/shared/opsutil/evm.go

```diff
@@ -0,0 +1,146 @@
+package opsutil
+
+import (
+	"fmt"
+	"math/big"
+
+	"github.com/Masterminds/semver/v3"
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	cldf "github.com/smartcontractkit/chainlink-deployments-framework/deployment"
+	"github.com/smartcontractkit/chainlink-deployments-framework/operations"
+	"github.com/smartcontractkit/chainlink/deployment/ccip/shared/deployergroup"
+	"github.com/smartcontractkit/chainlink/deployment/ccip/shared/stateview"
+	"github.com/smartcontractkit/chainlink/deployment/common/proposalutils"
+
+	mcmslib "github.com/smartcontractkit/mcms"
+	mcmssdk "github.com/smartcontractkit/mcms/sdk"
+	mcmstypes "github.com/smartcontractkit/mcms/types"
+)
+
+// Addressable is an interface that provides a method to get the address of a contract.
+type Addressable interface {
+	Address() common.Address
+}
+
+// EVMCallDeps contains the dependencies required for an EVM call operation.
+// It includes the contract bindings, the chain on which the operation is performed, and transaction options.
+type EVMCallDeps[C Addressable] struct {
+	Contract C
+	Chain    cldf.Chain
+	Opts     *bind.TransactOpts
+}
+
+// EVMCallInput is the input structure for an EVM call operation.
+// It contains the address of the contract, the chain selector, and the input data required for the call.
+// Why not pull the address from contract bindings and the chain selector from the chain? Because we need both to be included in the report.
+// This ensures we don't conflict with other operations that have the same call input but a different target address / target chain.
+type EVMCallInput[IN any] struct {
+	Address       common.Address
+	ChainSelector uint64
+	CallInput     IN
+}
+
+// EVMCallOutput is the output structure for an EVM call operation.
+// It contains the transaction that was sent and the type of contract that was called.
+type EVMCallOutput struct {
+	Tx           *types.Transaction
```

Comment:
nit: I think its better to use custom types/parser, geth usually cannot handle custom tx types (shouldnt be a problem for this particular scenario, may be except zkSync stack) but as a general practice